pynest/examples/Potjans_2014/network.py

```diff
@@ -160,6 +160,19 @@ def create_populations(self):
                     )
             self.DC_amp_e = compute_DC(self.net_dict, self.w_ext)
 
+        v0_type_options = ['optimized', 'original']
+        if self.net_dict['V0_type'] not in v0_type_options:
+            print(
+                '''
+                '{0}' is not a valid option for 'V0_type',
+                replacing it with '{1}.'
+                Valid options are {2}.
+                '''.format(self.net_dict['V0_type'],
+                           v0_type_options[0],
+                           v0_type_options)
+                )
+            self.net_dict['V0_type'] = v0_type_options[0]
```

Comment:
I don't like that it automatically chooses a value if the given value is invalid and I'd really prefer it to fail loudly with an error in this case. The way it is now, the network would in any case provide results, but their interpretation might be wrong as the initial conditions might be not the ones intended (if there was a typo, for instance).

I suggest to change it here and also in the master/2.20.1 version of the script, if that's the same.

Reply:
I agree, the warning will absolutely not be seen in the output. The concept should always be "fail noisy, succeed silently". Something unexpected should break the run, otherwise we couldn't trust the results, because we never know how many warnings we didn't see.

Reply:
I totally agree. This and other observations motivated me to [refactor the whole microcircuit](https://github.com/nest/nest-simulator/pull/1444). In my new implementation, an [Exception](https://github.com/nest/nest-simulator/blob/master/pynest/examples/Potjans_2014/network.py#L334) punishes wrong input.
However, since this is a backport, I assumed that it would be useful to include the whole [PR introducing the new initial conditions](https://github.com/nest/nest-simulator/pull/1260) and only change the algorithm.
But if you also think that this warning does more harm than good, I will just update it as well.

Reply:
I definitely am all in favour of changing this! Please go ahead!

But quite honestly, I'm not yet convinced of the purpose of backporting these kind of changes in the first place. Whom are we expecting to actually use a three year old release in practice? Let alone update it to get new results with it? In my humble opinion I would only backport real bug fixes, but leave all examples and such unchanged. If a user wants to do such a comparison, they could always do the backport themselves...

Reply:
BTW: thanks for linking to the corresponding code locations in such a nice way. It's really refreshing to see some Markdown being put to good use! :heart:

Reply:
Now, a wrong `V0_type` triggers an Exception.

The reason for this backport is that NEST 2.14 serves as reference for benchmarks with models like Potjans_2014.